File: javadevelopers-understanding-kotlin-properties/src/test/java/com/moelholm/jav/TwoTests.java
package com.moelholm.jav;

import static org.assertj.core.api.Java6Assertions.assertThat;

import com.moelholm.Two;
import org.junit.Test;

public class TwoTests {

  @Test
  public void test() {

    Two two = new Two("Hello from Java");

    assertThat(two.getMessage()).isEqualTo("Hello from Java");

    two.setMessage("Hello again from Java");

    assertThat(two.getMessage()).isEqualTo("Hello again from Java");

  }

}

File: javadevelopers-understanding-kotlin-properties/src/test/kotlin/com/moelholm/kot/TwoTests.kt
package com.moelholm.kot

import com.moelholm.Two
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class TwoTests {

  @Test
  fun test() {

    val two = Two("Hello from Kotlin")

    assertThat(two.message).isEqualTo("Hello from Kotlin")

    two.message = "Hello again from Kotlin"

    assertThat(two.message).isEqualTo("Hello again from Kotlin")

  }

}